public/app/features/logs/legacyLogsFrame.ts

```diff
@@ -0,0 +1,50 @@
+import { DataFrame, FieldCache, FieldType, Field } from '@grafana/data';
+
+import type { LogsFrame, Attributes } from './logsFrame';
+
+function getAttributes(frame: DataFrame, cache: FieldCache, lineField: Field): Attributes[] | undefined {
+  const useLabelsField = frame.meta?.custom?.frameType === 'LabeledTimeValues';
+
+  if (!useLabelsField) {
+    const lineLabels = lineField.labels;
+    if (lineLabels != null) {
+      const result = new Array(frame.length);
+      result.fill(lineLabels);
+      return result;
+    } else {
+      return undefined;
+    }
+  }
+
+  const labelsField = cache.getFieldByName('labels');
+
+  if (labelsField == null) {
+    return undefined;
+  }
+
+  return labelsField.values;
+}
+
+export function parseLegacyLogsFrame(frame: DataFrame): LogsFrame | null {
+  const cache = new FieldCache(frame);
+  const timeField = cache.getFields(FieldType.time)[0];
+  const bodyField = cache.getFields(FieldType.string)[0];
```

Comment:
Is this error safe? What if `cache.getFields(FieldType.string)` returns an empty array?

Reply:
it is checked on the next line:
```js
  // these two are mandatory
  if (timeField === undefined || bodyField === undefined) {
    return null;
  }
```

if `getFields` returns an empty-array, then any item from it will be `undefined`.


would you prefer the checking to happen more explicitly?

like:
```ts
const timeFields = cache.getFields(FieldType.time);
const bodyFields = cache.getFields(FieldType.string);

if (timeFields.length === 0 || bodyFields.length === 0) {
    return null;
}

const timeField = timeFields[0];
const bodyField = bodyFields[0];
```

?